### ISSUE TITLE
[Bug]: FileNotFoundError: Could not find module 'atiadlxx.dll' (or one of its dependencies).

### ISSUE THREAD

**Issue by StudioDUzes** [lshqqytiger/stable-diffusion-webui-directml#219]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits

### What happened?

WEBUI don't start after update...

### Steps to reproduce the problem

start webui...

### What should have happened?

start

### Version or Commit where the problem happens

Version: 1.5.1 Commit hash: 723974a8b74ffd5acc1a1fdaf36e9c2426651537

### What Python version are you running on ?

Python 3.10.x

### What platforms do you use to access the UI ?

Windows

### What device are you running WebUI on?

Other GPUs

### Cross attention optimization

Automatic

### What browsers do you use to access the UI ?

Google Chrome

### Command Line Arguments

```Shell
--device-id 1 --port 7861 --medvram --opt-sdp-no-mem-attention --opt-sdp-attention --no-half --no-half-vae --precision full --opt-sub-quad-attention --opt-split-attention-v1 --disable-nan-check --autolaunch
```

### List of extensions

Infinite-Zoom

### Console logs

```Shell
Already up to date.
venv "L:\stable-diffusion-webui-directml\venv\Scripts\Python.exe"
fatal: No names found, cannot describe anything.
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: 1.5.1
Commit hash: 723974a8b74ffd5acc1a1fdaf36e9c2426651537


Installing requirements 1 for Infinite-Zoom



Launching Web UI with arguments: --device-id 1 --port 7861 --medvram --opt-sdp-no-mem-attention --opt-sdp-attention --no-half --no-half-vae --precision full --opt-sub-quad-attention --opt-split-attention-v1 --disable-nan-check --autolaunch
no module 'xformers'. Processing without...
no module 'xformers'. Processing without...
No module 'xformers'. Proceeding without it.
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ L:\stable-diffusion-webui-directml\launch.py:39 in <module>                                      │
│                                                                                                  │
│   36                                                                                             │
│   37                                                                                             │
│   38 if __name__ == "__main__":                                                                  │
│ ❱ 39 │   main()                                                                                  │
│   40                                                                                             │
│                                                                                                  │
│ L:\stable-diffusion-webui-directml\launch.py:35 in main                                          │
│                                                                                                  │
│   32 │   if args.test_server:                                                                    │
│   33 │   │   configure_for_tests()                                                               │
│   34 │                                                                                           │
│ ❱ 35 │   start()                                                                                 │
│   36                                                                                             │
│   37                                                                                             │
│   38 if __name__ == "__main__":                                                                  │
│                                                                                                  │
│ L:\stable-diffusion-webui-directml\modules\launch_utils.py:443 in start                          │
│                                                                                                  │
│   440                                                                                            │
│   441 def start():                                                                               │
│   442 │   print(f"Launching {'API server' if '--nowebui' in sys.argv else 'Web UI'} with argum   │
│ ❱ 443 │   import webui                                                                           │
│   444 │   if '--nowebui' in sys.argv:                                                            │
│   445 │   │   webui.api_only()                                                                   │
│   446 │   else:                                                                                  │
│                                                                                                  │
│ L:\stable-diffusion-webui-directml\webui.py:54 in <module>                                       │
│                                                                                                  │
│    51 startup_timer.record("import ldm")                                                         │
│    52                                                                                            │
│    53 from modules import extra_networks                                                         │
│ ❱  54 from modules.call_queue import wrap_gradio_gpu_call, wrap_queued_call, queue_lock  # noq   │
│    55                                                                                            │
│    56 # Truncate version number of nightly/local build of PyTorch to not cause exceptions with   │
│    57 if ".dev" in torch.__version__ or "+git" in torch.__version__:                             │
│                                                                                                  │
│ L:\stable-diffusion-webui-directml\modules\call_queue.py:6 in <module>                           │
│                                                                                                  │
│     3 import threading                                                                           │
│     4 import time                                                                                │
│     5                                                                                            │
│ ❱   6 from modules import shared, progress, errors                                               │
│     7                                                                                            │
│     8 queue_lock = threading.Lock()                                                              │
│     9                                                                                            │
│                                                                                                  │
│ L:\stable-diffusion-webui-directml\modules\shared.py:72 in <module>                              │
│                                                                                                  │
│    69 if cmd_opts.olive:                                                                         │
│    70 │   cmd_opts.onnx = True                                                                   │
│    71 if cmd_opts.backend == "directml":                                                         │
│ ❱  72 │   directml_init()                                                                        │
│    73                                                                                            │
│    74                                                                                            │
│    75 devices.device, devices.device_interrogate, devices.device_gfpgan, devices.device_esrgan   │
│                                                                                                  │
│ L:\stable-diffusion-webui-directml\modules\dml\__init__.py:38 in directml_init                   │
│                                                                                                  │
│   35 │   torch.cuda.mem_get_info = torch.dml.mem_get_info                                        │
│   36                                                                                             │
│   37 def directml_init():                                                                        │
│ ❱ 38 │   from modules.dml.backend import DirectML # pylint: disable=ungrouped-imports            │
│   39 │   # Alternative of torch.cuda for DirectML.                                               │
│   40 │   torch.dml = DirectML                                                                    │
│   41                                                                                             │
│                                                                                                  │
│ L:\stable-diffusion-webui-directml\modules\dml\backend.py:10 in <module>                         │
│                                                                                                  │
│    7 from .utils import rDevice, get_device                                                      │
│    8 from .device import device                                                                  │
│    9 from .device_properties import DeviceProperties                                             │
│ ❱ 10 from .memory_amd import AMDMemoryProvider                                                   │
│   11 from .memory import MemoryProvider                                                          │
│   12                                                                                             │
│   13 def amd_mem_get_info(device: Optional[rDevice]=None) -> tuple[int, int]:                    │
│                                                                                                  │
│ L:\stable-diffusion-webui-directml\modules\dml\memory_amd\__init__.py:1 in <module>              │
│                                                                                                  │
│ ❱ 1 from .driver.atiadlxx import ATIADLxx                                                        │
│   2                                                                                              │
│   3 class AMDMemoryProvider:                                                                     │
│   4 │   driver: ATIADLxx = ATIADLxx()                                                            │
│                                                                                                  │
│ L:\stable-diffusion-webui-directml\modules\dml\memory_amd\driver\atiadlxx.py:2 in <module>       │
│                                                                                                  │
│    1 import ctypes as C                                                                          │
│ ❱  2 from .atiadlxx_apis import *                                                                │
│    3 from .atiadlxx_structures import *                                                          │
│    4 from .atiadlxx_defines import *                                                             │
│    5                                                                                             │
│                                                                                                  │
│ L:\stable-diffusion-webui-directml\modules\dml\memory_amd\driver\atiadlxx_apis.py:6 in <module>  │
│                                                                                                  │
│    3 from .atiadlxx_structures import *                                                          │
│    4                                                                                             │
│    5 if 'Windows' in platform():                                                                 │
│ ❱  6 │   atiadlxx = C.WinDLL("atiadlxx.dll")                                                     │
│    7 else:                                                                                       │
│    8 │   atiadlxx = C.CDLL("libatiadlxx.so") # Not tested on Linux system. But will be suppor    │
│    9                                                                                             │
│                                                                                                  │
│ C:\Users\StudioD\AppData\Local\Programs\Python\Python310\lib\ctypes\__init__.py:374 in __init__  │
│                                                                                                  │
│   371 │   │   self._FuncPtr = _FuncPtr                                                           │
│   372 │   │                                                                                      │
│   373 │   │   if handle is None:                                                                 │
│ ❱ 374 │   │   │   self._handle = _dlopen(self._name, mode)                                       │
│   375 │   │   else:                                                                              │
│   376 │   │   │   self._handle = handle                                                          │
│   377                                                                                            │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
FileNotFoundError: Could not find module 'atiadlxx.dll' (or one of its dependencies). Try using the full path with
constructor syntax.
Appuyez sur une touche pour continuer...
```

### Additional information

GPU Intel arc a770 16go

**Comment by Nathan-dm** [comment]
i got same error after update 
(GPU Intel Xe Max 4gb)

**Comment by Nathan-dm** [comment]
> Change `DirectML memory stats provider` to `Performance Counter` or `None` in Settings -> Optimization.

how do i change? even webui didn't launch

**Comment by StudioDUzes** [comment]
"cross_attention_optimization": "None", in config.json don't work...

**Comment by StudioDUzes** [comment]
> Change `DirectML memory stats provider` to `Performance Counter` or `None` in Settings -> Optimization.

Which file for  `DirectML memory stats provider` to `Performance Counter` ?

"cross_attention_optimization": "None", in config.json don't work...

**Comment by lshqqytiger** [comment]
I already added GPU check to avoid this error. I don't know why it does not work as intended.
Add `"directml_memory_provider": "Performance Counter"` to `config.json` like below.
```json
    ...
    "canvas_blur_prompt": false,
    "directml_memory_provider": "Performance Counter"
}
```

**Comment by Nathan-dm** [comment]
> I already added GPU check to avoid this error. I don't know why it does not work as intended. Add `"directml_memory_provider": "Performance Counter"` to `config.json` like below.
> 
> ```json
>     ...
>     "canvas_blur_prompt": false,
>     "directml_memory_provider": "Performance Counter"
> }
> ```

still didn't work 
<details><summary>my config.json</summary>
<p>

```json
{
    "samples_save": true,
    "samples_format": "png",
    "samples_filename_pattern": "",
    "save_images_add_number": true,
    "grid_save": true,
    "grid_format": "png",
    "grid_extended_filename": false,
    "grid_only_if_multiple": true,
    "grid_prevent_empty_spots": false,
    "n_rows": -1,
    "enable_pnginfo": true,
    "save_txt": false,
    "save_images_before_face_restoration": false,
    "save_images_before_highres_fix": false,
    "save_images_before_color_correction": false,
    "save_mask": false,
    "save_mask_composite": false,
    "jpeg_quality": 80,
    "webp_lossless": false,
    "export_for_4chan": true,
    "img_downscale_threshold": 4.0,
    "target_side_length": 4000,
    "img_max_size_mp": 200,
    "use_original_name_batch": true,
    "use_upscaler_name_as_suffix": false,
    "save_selected_only": true,
    "save_init_img": false,
    "temp_dir": "",
    "clean_temp_dir_at_start": false,
    "outdir_samples": "",
    "outdir_txt2img_samples": "outputs/txt2img-images",
    "outdir_img2img_samples": "outputs/img2img-images",
    "outdir_extras_samples": "outputs/extras-images",
    "outdir_grids": "",
    "outdir_txt2img_grids": "outputs/txt2img-grids",
    "outdir_img2img_grids": "outputs/img2img-grids",
    "outdir_save": "log/images",
    "outdir_init_images": "outputs/init-images",
    "save_to_dirs": true,
    "grid_save_to_dirs": true,
    "use_save_to_dirs_for_ui": false,
    "directories_filename_pattern": "[date]",
    "directories_max_prompt_words": 8,
    "ESRGAN_tile": 192,
    "ESRGAN_tile_overlap": 8,
    "realesrgan_enabled_models": [
        "R-ESRGAN 4x+",
        "R-ESRGAN 4x+ Anime6B"
    ],
    "upscaler_for_img2img": null,
    "face_restoration_model": "CodeFormer",
    "code_former_weight": 0.5,
    "face_restoration_unload": false,
    "show_warnings": false,
    "memmon_poll_rate": 8,
    "samples_log_stdout": false,
    "multiple_tqdm": false,
    "print_hypernet_extra": false,
    "list_hidden_files": true,
    "unload_models_when_training": false,
    "pin_memory": false,
    "save_optimizer_state": false,
    "save_training_settings_to_txt": true,
    "dataset_filename_word_regex": "",
    "dataset_filename_join_string": " ",
    "training_image_repeats_per_epoch": 1,
    "training_write_csv_every": 500,
    "training_xattention_optimizations": false,
    "training_enable_tensorboard": false,
    "training_tensorboard_save_images": false,
    "training_tensorboard_flush_every": 120,
    "sd_model_checkpoint": "dreamshaper_631BakedVae.safetensors [853e7e85c4]",
    "sd_checkpoint_cache": 0,
    "sd_vae_checkpoint_cache": 0,
    "sd_vae": "Automatic",
    "sd_vae_as_default": true,
    "inpainting_mask_weight": 1.0,
    "initial_noise_multiplier": 1.0,
    "img2img_color_correction": false,
    "img2img_fix_steps": false,
    "img2img_background_color": "#ffffff",
    "enable_quantization": true,
    "enable_emphasis": true,
    "enable_batch_seeds": true,
    "comma_padding_backtrack": 20,
    "CLIP_stop_at_last_layers": 7,
    "upcast_attn": true,
    "randn_source": "GPU",
    "cross_attention_optimization": "sub-quadratic",
    "s_min_uncond": 0,
    "token_merging_ratio": 0.5,
    "token_merging_ratio_img2img": 0.0,
    "token_merging_ratio_hr": 0.0,
    "use_old_emphasis_implementation": false,
    "use_old_karras_scheduler_sigmas": false,
    "no_dpmpp_sde_batch_determinism": false,
    "use_old_hires_fix_width_height": false,
    "dont_fix_second_order_samplers_schedule": false,
    "interrogate_keep_models_in_memory": false,
    "interrogate_return_ranks": false,
    "interrogate_clip_num_beams": 1,
    "interrogate_clip_min_length": 24,
    "interrogate_clip_max_length": 48,
    "interrogate_clip_dict_limit": 1500,
    "interrogate_clip_skip_categories": [],
    "interrogate_deepbooru_score_threshold": 0.5,
    "deepbooru_sort_alpha": true,
    "deepbooru_use_spaces": true,
    "deepbooru_escape": true,
    "deepbooru_filter_tags": "",
    "extra_networks_show_hidden_directories": true,
    "extra_networks_hidden_models": "When searched",
    "extra_networks_default_view": "cards",
    "extra_networks_default_multiplier": 1.0,
    "extra_networks_card_width": 0,
    "extra_networks_card_height": 0,
    "extra_networks_add_text_separator": " ",
    "ui_extra_networks_tab_reorder": "",
    "sd_hypernetwork": "None",
    "localization": "None",
    "gradio_theme": "Default",
    "img2img_editor_height": 720,
    "return_grid": true,
    "return_mask": false,
    "return_mask_composite": false,
    "do_not_show_images": false,
    "send_seed": true,
    "send_size": true,
    "font": "",
    "js_modal_lightbox": true,
    "js_modal_lightbox_initially_zoomed": true,
    "js_modal_lightbox_gamepad": false,
    "js_modal_lightbox_gamepad_repeat": 250,
    "show_progress_in_title": true,
    "samplers_in_dropdown": true,
    "dimensions_and_batch_together": true,
    "keyedit_precision_attention": 0.1,
    "keyedit_precision_extra": 0.05,
    "keyedit_delimiters": ".,\\/!?%^*;:{}=`~()",
    "quicksettings_list": [
        "sd_model_checkpoint"
    ],
    "ui_tab_order": [],
    "hidden_tabs": [],
    "ui_reorder": "inpaint, sampler, checkboxes, hires_fix, dimensions, cfg, seed, batch, override_settings, scripts",
    "hires_fix_show_sampler": false,
    "hires_fix_show_prompts": false,
    "add_model_hash_to_info": true,
    "add_model_name_to_info": true,
    "add_version_to_infotext": true,
    "disable_weights_auto_swap": true,
    "show_progressbar": false,
    "live_previews_enable": true,
    "live_previews_image_format": "png",
    "show_progress_grid": true,
    "show_progress_every_n_steps": 4,
    "show_progress_type": "Approx NN",
    "live_preview_content": "Prompt",
    "live_preview_refresh_period": 1000,
    "hide_samplers": [],
    "eta_ddim": 0.0,
    "eta_ancestral": 1.0,
    "ddim_discretize": "uniform",
    "s_churn": 0.0,
    "s_tmin": 0.0,
    "s_noise": 1.0,
    "eta_noise_seed_delta": 0,
    "always_discard_next_to_last_sigma": false,
    "uni_pc_variant": "bh1",
    "uni_pc_skip_type": "time_uniform",
    "uni_pc_order": 3,
    "uni_pc_lower_order_final": true,
    "postprocessing_enable_in_main_ui": [],
    "postprocessing_operation_order": [],
    "upscaling_max_images_in_cache": 5,
    "disabled_extensions": [],
    "disable_all_extensions": "none",
    "restore_config_state_file": "",
    "sd_checkpoint_hash": "853e7e85c46be9ad4815666cd335545df32dab5e0d22afe5ccc6ad9bda834b52",
    "ldsr_steps": 100,
    "ldsr_cached": false,
    "SCUNET_tile": 256,
    "SCUNET_tile_overlap": 8,
    "SWIN_tile": 192,
    "SWIN_tile_overlap": 8,
    "lora_functional": false,
    "sd_lora": "None",
    "lora_preferred_name": "Alias from file",
    "lora_add_hashes_to_infotext": true,
    "ui_reorder_list": [
        "inpaint",
        "sampler",
        "checkboxes",
        "hires_fix",
        "dimensions",
        "cfg",
        "seed",
        "batch",
        "override_settings",
        "scripts"
    ],
    "grid_zip_filename_pattern": "",
    "grid_text_active_color": "#000000",
    "grid_text_inactive_color": "#999999",
    "grid_background_color": "#ffffff",
    "disable_mmap_load_safetensors": false,
    "sd_unet": "Automatic",
    "auto_vae_precision": false,
    "sdxl_crop_top": 0.0,
    "sdxl_crop_left": 0.0,
    "sdxl_refiner_low_aesthetic_score": 2.5,
    "sdxl_refiner_high_aesthetic_score": 6.0,
    "enable_mem_pattern": true,
    "enable_mem_reuse": true,
    "reload_model_before_each_generation": false,
    "offload_state_dict": false,
    "pad_cond_uncond": false,
    "experimental_persistent_cond_cache": false,
    "hires_fix_use_firstpass_conds": false,
    "extra_networks_card_text_scale": 1,
    "extra_networks_card_show_desc": true,
    "textual_inversion_print_at_load": false,
    "textual_inversion_add_hashes_to_infotext": true,
    "sd_lyco": "None",
    "lora_show_all": false,
    "lora_hide_unknown_for_versions": [],
    "keyedit_move": true,
    "disable_token_counters": false,
    "extra_options": [],
    "extra_options_accordion": false,
    "add_user_name_to_info": false,
    "infotext_styles": "Apply if any",
    "k_sched_type": "Automatic",
    "sigma_min": 0.0,
    "sigma_max": 0.0,
    "rho": 0.0,
    "canvas_hotkey_zoom": "Alt",
    "canvas_hotkey_adjust": "Ctrl",
    "canvas_hotkey_move": "F",
    "canvas_hotkey_fullscreen": "S",
    "canvas_hotkey_reset": "R",
    "canvas_hotkey_overlap": "O",
    "canvas_show_tooltip": true,
    "canvas_blur_prompt": false,
    "directml_memory_provider": "Performance Counter",
    "canvas_disabled_functions": [
        "Overlap"
    ]
    
}


``` 

</p>
</details>

**Comment by lshqqytiger** [comment]
May be fixed in 3f69a46aced47d8b14a9abb24181231ae7976966

**Comment by StudioDUzes** [comment]
It is ok for me with :

"canvas_blur_prompt": false,
    "directml_memory_provider": "Performance Counter",

in config.json

Thank you very much

Already up to date.
venv "L:\stable-diffusion-webui-directml\venv\Scripts\Python.exe"
fatal: No names found, cannot describe anything.
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: 1.5.1
Commit hash: 3f69a46aced47d8b14a9abb24181231ae7976966
Launching Web UI with arguments: --device-id 1 --port 7861 --medvram --opt-sdp-no-mem-attention --opt-sdp-attention --no-half --no-half-vae --precision full --opt-sub-quad-attention --opt-split-attention-v1 --disable-nan-check --autolaunch
no module 'xformers'. Processing without...
no module 'xformers'. Processing without...
No module 'xformers'. Proceeding without it.
Warning: caught exception 'Something went wrong.', memory monitor disabled
Loading weights [31e35c80fc] from L:\stable-diffusion-webui-directml\models\Stable-diffusion\SDXL\sd_xl_base_1.0.safetensors
Running on local URL:  http://127.0.0.1:7861

To create a public link, set `share=True` in `launch()`.
Startup time: 11.0s (launcher: 0.5s, import torch: 3.6s, import gradio: 1.5s, setup paths: 0.8s, other imports: 1.8s, opts onchange: 0.3s, list SD models: 0.2s, load scripts: 0.9s, create ui: 0.6s, gradio launch: 0.7s).
Creating model from config: L:\stable-diffusion-webui-directml\repositories\generative-models\configs\inference\sd_xl_base.yaml
Applying attention optimization: sdp-no-mem... done.
Model loaded in 20.7s (load weights from disk: 1.8s, create model: 0.5s, apply weights to model: 9.9s, calculate empty prompt: 8.6s).

**Comment by StudioDUzes** [comment]
> May be fixed in [3f69a46](https://github.com/lshqqytiger/stable-diffusion-webui-directml/commit/3f69a46aced47d8b14a9abb24181231ae7976966)

It is ok for me on new installation...

venv "L:\stable-diffusion-webui-directml\venv\Scripts\Python.exe"
fatal: No names found, cannot describe anything.
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: 1.5.1
Commit hash: 3f69a46aced47d8b14a9abb24181231ae7976966
Launching Web UI with arguments:
no module 'xformers'. Processing without...
no module 'xformers'. Processing without...
No module 'xformers'. Proceeding without it.
Loading weights [6ce0161689] from L:\stable-diffusion-webui-directml\models\Stable-diffusion\v1-5-pruned-emaonly.safetensors
Creating model from config: L:\stable-diffusion-webui-directml\configs\v1-inference.yaml
LatentDiffusion: Running in eps-prediction mode
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Startup time: 7.6s (launcher: 0.4s, import torch: 2.8s, import gradio: 0.8s, setup paths: 0.6s, other imports: 1.3s, load scripts: 0.8s, create ui: 0.5s, gradio launch: 0.2s).
DiffusionWrapper has 859.52 M params.
Applying attention optimization: InvokeAI... done.
Model loaded in 4.0s (load weights from disk: 0.7s, create model: 0.4s, apply weights to model: 0.8s, apply half(): 0.6s, move model to device: 1.1s, calculate empty prompt: 0.4s).
